hugr-core/src/builder/dataflow.rs

```diff
@@ -231,10 +152,10 @@ impl FunctionBuilder<Hugr> {
         }
 
         // Update the builder metadata
-        self.0.num_out_wires += 1;
+        self.num_out_wires += 1;
     }
 
-    /// Update the function builder's parent signature.
+    /// Update the DFG builder's parent signature.
     ///
     /// Internal function used in [add_input] and [add_output].
     ///
```

Comment:
`update_fn_signature` still needs to be updated or it will panic for anything other than `FuncDefn`.

Reply:
Yeah, so you've moved `add_input` and `add_output` from  `FunctionBuilder<Hugr>` into `DFGBuilder<Hugr>` and a proxy for that in `FunctionBuilder<Hugr>`. Instead you could define
```
trait MutableIO : Dataflow {
  fn update_signature(...); // no default impl
  fn add_input(...) { .... } // code you've moved becomes default impl delegating to self.update_signature
  fn add_output(...) { ....} // similarly
}
```
then you `impl MutableIO for FunctionBuilder<Hugr>` and `impl MutableIO for DFGBuilder<Hugr>` and just have to provide the two `update_signature` methods, i.e. then you don't need the `delegate! { to self.0 { fn add_input(...); fn add_output(...); } }`. Does that work??